src/syntax.ml

```diff
@@ -59,7 +63,8 @@ and p_type = p_term
 and p_patt = p_term
 
 (** Parser-level representation of a function argument. *)
-and p_arg = ident * p_type option
+(*  the boolean tells whether this argument is implicit *)
+and p_arg = ident * p_type option * bool
```

Comment:
Why not replacing bool by implicitness? Or, conversely, why not replacing implicitness by bool?

Reply:
After all, sym_implicits use bool too.

Reply:
Yes, use `bool` everywhere, it is easier to match on.